astropy/io/fits/header.py

```diff
@@ -1423,8 +1423,7 @@ def insert(self, key, card, useblanks=True, after=False):
         # used by list.insert()
         if idx < 0:
             idx += len(self._cards) - 1
-            if idx < 0:
-                idx = 0
+            idx = max(idx, 0)
```

Comment:
Yikes both to having `+= len(self._cards) -1` (why the `-1`??) and to making everything too low 0.  But that feels very out of scope here... (without it, one could just use [`normalize_axis_index`](https://numpy.org/doc/stable/reference/generated/numpy.lib.array_utils.normalize_axis_index.html))